### PR TITLE
[886] Implement slugs on degrees

### PR DESCRIPTION
**Pull request by darokel**
### Context

- https://trello.com/c/bVWfzF3w/886-add-slugs-to-degrees

### Changes proposed in this pull request

- Build on the work done in this PR https://github.com/DFE-Digital/register-trainee-teachers/pull/405/files to add slugs to degrees
- Note, we've taken out the immutability changes out of this PR and will add to a separate PR right after this one. This is so we can run the migrations to set existing records which may be in the QA environment then enforce the readonly mode in subsequent PR otherwise the data changes in the migrations in this PR won't change anything.

### Guidance to review

Screenshot 2021-01-20 at 15 36 04

### CODE REVIEW

**Comment by darokel** [comment]
[Add slugs to degrees](https://trello.com/c/bVWfzF3w/886-add-slugs-to-degrees)

File: app/controllers/trainees/degrees_controller.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ def new
 
     def edit
       authorize trainee
-      @degree = trainee.degrees.find(params[:id])
+      @degree = trainee.degrees.find_by(slug: params[:id])
```
Comment (AbigailMcP) [inline app/controllers/trainees/degrees_controller.rb:12]: Can all these calls to `find_by` be our sluggable `from_param`?

**Review by AbigailMcP — COMMENTED** [review]
One small comment, otherwise looks good!

Couple of failing tests due to passing in the degree id to access a route.

File: app/controllers/trainees/degrees_controller.rb (hunk 1 above)
Comment (darokel) [reply app/controllers/trainees/degrees_controller.rb]: yup and done! https://github.com/DFE-Digital/register-trainee-teachers/pull/411/commits/ec826dba2e4813cf46436aec0706d4ea85b26527

**Review by AbigailMcP — APPROVED** [review]
Looks good!

Let's 🚢 once the tests have passed.

**Pull request merged**